oximeter/collector/src/lib.rs

```diff
@@ -202,10 +247,32 @@ impl Oximeter {
         }
         info!(log, "starting oximeter server");
 
-        let resolver = Resolver::new_from_ip(
-            log.new(o!("component" => "DnsResolver")),
-            *args.address.ip(),
-        )?;
+        // Use the address for Oximeter to infer the bootstrap DNS address
+        let bootstrap_dns: Vec<SocketAddr> =
+            get_internal_dns_server_addresses(*args.address.ip())
+                .into_iter()
+                .map(|ip| SocketAddr::new(ip, DNS_PORT))
+                .collect();
+
+        let make_clickhouse_resolver = || -> BoxedResolver {
+            if let Some(address) = config.db.address {
+                Box::new(SingleHostResolver::new(address))
+            } else {
+                let service = if config.db.replicated {
+                    ServiceName::ClickhouseServer
+                } else {
+                    ServiceName::Clickhouse
+                };
+                Box::new(DnsResolver::new(
+                    service::Name(service.srv_name()),
+                    bootstrap_dns.clone(),
+                    DnsResolverConfig {
+                        hardcoded_ttl: Some(tokio::time::Duration::MAX),
+                        ..Default::default()
+                    },
```

Comment:
The configurations are defined here: https://docs.rs/qorb/0.0.1/qorb/resolvers/dns/struct.DnsResolverConfig.html

Although it's supported in qorb, I haven't yet enabled "dynamic DNS resolution" (e.g., if you get new DNS servers, do we find them from our initial set of DNS servers, and start querying them too?). I haven't yet confirmed that DNS servers actually have SRV records for themselves...

Also: I'm *pretty* sure we set the value of "TTL" to zero right now, so I'm ignoring it. We still re-query DNS on a regular interval, and should swap backends if records change definitively, but there won't be auto-eviction of old backends.